### PR TITLE
Jenkins minor tweaks

### PR DESCRIPTION
**Pull request by gerhardol**
Part of #5038

Changes proposed in this pull request:
- Minor changes to Jenkins plugin, checks
 
Screenshots before and after (if PR changes UI):
-n/a

What did I do to test the code and ensure quality:
- Manual tests, Mono or builds.apache.org has public Jenkins servers (also multibranch pipeline)

Has been tested on (remove any that don't apply):
- Windows 7 and 10

### CODE REVIEW

File: Plugins/BuildServerIntegration/JenkinsIntegration/JenkinsAdapter.cs [hunk 1]
```diff
@@ -267,13 +270,24 @@ private void ObserveBuilds(DateTime? sinceDate, bool? running, IObserver<BuildIn
...
 
-                        var buildInfo = CreateBuildInfo((JObject)buildDetails);
-                        observer.OnNext(buildInfo);
-
-                        if (buildInfo.Status == BuildInfo.BuildStatus.InProgress)
+                        try
+                        {
+                            var buildInfo = CreateBuildInfo((JObject)buildDetails);
+
+                            if (buildInfo != null)
```
Comment (gerhardol) [inline Plugins/BuildServerIntegration/JenkinsIntegration/JenkinsAdapter.cs]: Null check is not strictly required

**Comment by vbjay** [comment]
Does it hurt?

File: Plugins/BuildServerIntegration/JenkinsIntegration/JenkinsAdapter.cs (hunk 1 above)
Comment (gerhardol) [reply Plugins/BuildServerIntegration/JenkinsIntegration/JenkinsAdapter.cs]: > Does it hurt?

No. But too many null checks can affect the readability. With ReSharper, you can see this better (I do not have a license.)

Build info is in this version currently set (try-catch added if failure parsing). This is a preparation for another PR that I have not decided if I will submit. I kept it as "it does not hurt".

Comment (RussKie) [reply Plugins/BuildServerIntegration/JenkinsIntegration/JenkinsAdapter.cs]: That's why it is preferable to invert `if`s and reduce nesting.
We'll get you a licence in few months time when the renewal time comes 
/cc: @KindDragon

**Comment by RussKie** [comment]
LGTM

**Pull request merged**